plugin/core/windows.py

```diff
@@ -557,6 +566,43 @@ def _check_window_closed(self) -> None:
         if not self._is_closing and not self._window.is_valid():
             self._handle_window_closed()
 
+    def _receive_progress_token(self, params: Dict[str, Any], client: Client, request_id: Any) -> None:
+        self._progress[params['token']] = dict()
+        client.send_response(Response(request_id, None))
+
+    def _handle_progress_notification(self, params: Dict[str, Any]) -> None:
+        token = params['token']
+        if token not in self._progress:
+            debug('unknown $/progress token: {}'.format(token))
+            return
+        value = params['value']
+        if not value:
+            return
```

Comment:
What would be the point of some server sending no (or null) value? That sounds like a server issue and probably shouldn't just be ignored?

Reply:
Also, non-false'y value doesn't guarantee that it's a Dict so it would crash later anyway.

Reply:
Well, this is just to avoid errors if the server doesn't follow the spec. Actually I just was following the style of the PR #902 here (https://github.com/sublimelsp/LSP/blob/06bdc86a980d27ac24989ab7daff9398493f5a9a/plugin/core/windows.py#L647).

I'm not sure if we can assume all servers to follow the specs everywhere and just ignore potential error sources. In fact, I recently submitted a small change here in #1004 because the Julia server sends a null value for a diagnostics related item, which resulted in an error in the Sublime console and broke the LSP diagnostics functionality completely. Actually, I think null wasn't even allowed in the specs for that value, so it was the servers fault, but now we are on the safe side for all servers.

Reply:
For diagnostics, we should be somewhat careful. But for straightforward notification handlers, these things are wrapped in a giant try ... except block, so any error checking is sort of redundant.

Reply:
I was thinking in terms of server developers (having in mind today's discussion on discord) which might want to have some indication if something goes wrong rather than silently ignoring the issue.

But in the end it's a design decision and for project owners to decide. But then @rwols has good points that it should be needed here as it won't break LSP.